core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,70 @@
+{% macro get_show_grant_sql(relation) %}
+{{ return(adapter.dispatch("get_show_grant_sql", "dbt")(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+show grants on {{ relation.type }} {{ relation }}
+{% endmacro %}
+
+{% macro get_grant_sql(relation, grant_config) %}
+{{ return(adapter.dispatch('get_grant_sql', 'dbt')(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_grant_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+        {% set grantees = grant_config[privilege] %}
+        {% if grantees %}
+            {% for grantee in grantees %}
+                grant {{ privilege }} on {{ relation.type }} {{ relation }} to {{ grantee}};
+            {% endfor %}
+        {% endif %}
+    {% endfor %}
+{% endmacro %}
+
+{% macro get_revoke_sql(relation, grant_config) %}
+{{ return(adapter.dispatch("get_revoke_sql", "dbt")(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_revoke_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+        {% set grantees = [] %}
+        {% set all_grantees = grant_config[privilege] %}
+        {% for grantee in all_grantees %}
+            {% if grantee != target.user %}
```

Comment:
IMO the `target.user` filter would be better in post-processing / standardization, rather than here, since this behavior will differ by adapter (= whether the filter can happen in SQL or not).

To make that possible, I think we could either:
1. Reach deep into the `adapter` object to find `credentials.user` or `credentials.role`
2. Add an argument to `standardize_grants_dict` which accepts `target.user` (default) / `target.role` (Snowflake)
3. Run a separate query that returns `current_user` / `current_role`

1 is gross, but it does nicely account for this being a potentially different property on different databases